validation/validation.js

```diff
@@ -0,0 +1,14 @@
+import Exception from "../exception/exception.js";
+
+export const validateRequest = (schema) => (req, res, next) => {
+    let error;
+    if (Object.keys(req.body).length === 0){
+        error = schema.validate(req.query);
```

Comment:
joi should throw exception if required keys are missing

Reply:
const { error } = req.method === 'GET' ? schema.validate(req.query) : schema.validate(req.body);

Reply:
done